rpc/wrtc_client_channel_test.go

```diff
@@ -538,3 +538,53 @@ func TestWebRTCClientChannelWithInterceptor(t *testing.T) {
 
 	wg.Wait()
 }
+
+func TestWebRTCClientChannelCanStopStreamRecvMsg(t *testing.T) {
```

Comment:
This test sets up a client-server connection, sends a message from the client, does not respond from the server, closes the underlying client peer connection, and tries to `RecvMsg` from the client side. It passes with the changes in this PR, and hangs forever on `main`.

I don't think it's a _perfect_ representation of the issue, but it's probably close enough.

Reply:
cool, i guess the main thing is does this result in a hang when we remove the new `select case` we added?